pkg/client/restclient/request.go

```diff
@@ -1090,6 +1090,17 @@ func (r Result) Into(obj runtime.Object) error {
 	return runtime.DecodeInto(r.decoder, r.body, obj)
 }
 
+// Decode decodes the result to a runtime.Object as determined by the decoder.
+func (r Result) Decode() (obj runtime.Object, gvk *unversioned.GroupVersionKind, err error) {
```

Comment:
Yes, how is this different than Get?

If it's important to get the gvk, you could modify the codec/serializer to just set the gvk in the object.